e2e/gallery.spec.ts

```diff
@@ -0,0 +1,465 @@
+    }
+  }
+});
+
+test("gallery page - pagination and infinite scroll", async ({ page }) => {
+  // 1. Navigate to the index page first to get available tables
+  await page.goto("/");
+
+  // 2. Wait for the page heading to become visible
+  await expect(
+    page.getByRole("heading", { name: /available views/i }),
+  ).toBeVisible();
+
+  // 3. Find gallery links
+  const galleryLinks = page.getByRole("link", { name: /gallery/i });
+  const linkCount = await galleryLinks.count();
+
+  if (linkCount > 0) {
+    // 4. Click the first gallery link
+    const firstGalleryLink = galleryLinks.first();
+    await firstGalleryLink.click();
+
+    // 5. Wait for the gallery page to load
+    await page.waitForURL("**/gallery/**", { timeout: 5000 });
+
+    // 6. Wait for the gallery container to be present
+    await page
+      .locator("#galleryContainer")
+      .waitFor({ state: "attached", timeout: 5000 });
+
+    // 7. Get initial item count
+    const initialItems = page.locator(".rounded-lg.border.bg-card");
+    const initialCount = await initialItems.count();
+
+    // 8. Scroll to bottom to trigger infinite scroll
+    await page.evaluate(() => {
+      window.scrollTo(0, document.body.scrollHeight);
+    });
+
+    // 9. Wait for potential new items to load
+    await page.waitForTimeout(2000);
+
+    // 10. Get new item count
+    const newItems = page.locator(".rounded-lg.border.bg-card");
+    const newCount = await newItems.count();
+
+    // 11. Verify either more items loaded or we're at the end
+    expect(newCount).toBeGreaterThanOrEqual(initialCount);
+  }
+});
+
+test("gallery page - data feature information display", async ({ page }) => {
+  // 1. Navigate to the index page first to get available tables
+  await page.goto("/");
+
+  // 2. Wait for the page heading to become visible
+  await expect(
+    page.getByRole("heading", { name: /available views/i }),
+  ).toBeVisible();
+
+  // 3. Find gallery links
+  const galleryLinks = page.getByRole("link", { name: /gallery/i });
+  const linkCount = await galleryLinks.count();
+
+  if (linkCount > 0) {
+    // 4. Click the first gallery link
+    const firstGalleryLink = galleryLinks.first();
+    await firstGalleryLink.click();
+
+    // 5. Wait for the gallery page to load
+    await page.waitForURL("**/gallery/**", { timeout: 5000 });
+
+    // 6. Wait for the gallery container to be present
+    await page
+      .locator("#galleryContainer")
+      .waitFor({ state: "attached", timeout: 5000 });
+
+    // 7. Get the first gallery item
+    const firstItem = page.locator(".rounded-lg.border.bg-card").first();
+    await expect(firstItem).toBeVisible();
+
+    // 8. Check for data source heading
+    const dataSourceHeading = firstItem.locator("h1");
+    if ((await dataSourceHeading.count()) > 0) {
+      await expect(dataSourceHeading).toBeVisible();
+    }
+
+    // 9. Check for feature information fields
+    const featureFields = firstItem.locator(".text-sm.font-medium");
```

Comment:
Same as above: let's add a class to the code to improve the ability to isolate the element.